pkg/cloudprovider/aws/apis/v1alpha1/provider.go

```diff
@@ -34,8 +34,8 @@ type AWS struct {
 	// TypeMeta includes version and kind of the extensions, inferred if not provided.
 	// +optional
 	metav1.TypeMeta `json:",inline"`
-	// InstanceProfile is the AWS identity that instances use.
-	// +required
+	// InstanceProfile is the AWS identity that instances use. Required if no default is configured on the controller.
```

Comment:
Thoughts on requiring the defaultInstanceProfile to be set. I think that semantic is a little more easily consumable.